bot/extensions/vote_cog.py

```diff
@@ -0,0 +1,315 @@
+from typing import Callable
+from discord.ext.commands import Cog, Context, hybrid_command, hybrid_group
+from bot.extensions.command_error_handler import send_command_help
+from collections.abc import Sequence
+from discord import Embed
+from discord.ui import Button, View
+import discord
+import random
+import asyncio
+
+
+# Problem: The modal doesn't allow more than 5 input fields (In the poll I assume there could be more)
+# Problem: Cannot make the bot output one modal after another (I can send 1 modal but after it says interaction expired or something).
+# When I try using webhooks it gives me 'Unknown Webhook' error
+# TODO(LIMITATION): Try to find a workaround with models. Or just accept user input: https://stackoverflow.com/questions/64949395/how-can-i-make-my-discord-bot-respond-with-a-follow-up-message
+
+
+#class TitleBuilder(ui.Modal):
+# 	def __init__(self):
+# 		super().__init__(title='Title')
+# 		self.input = ui.TextInput(label='Text')
+# 		self.add_item(self.input)
+
+# 	async def on_submit(self, interaction: discord.Interaction):
+# 		await interaction.response.send_message(f'Text: {self.input}', ephemeral=True)
+
+# class OptionBuilder(ui.Modal):
+# 	def __init__(self, option: int):
+# 		super().__init__(title=f'Option {option}')
+# 		self.option = ui.TextInput(label=f"Option {option}")
+# 		self.add_item(self.option)
+
+# 	async def on_submit(self, interaction: discord.Interaction):
+# 		await interaction.response.send_message(f'Text: {self.option}', ephemeral=True)
+
+class PollView(View):
+	def __init__(self, emojis: list[str], possible_emojis_size: int, ref_embed: Embed) -> None:
+		super().__init__()
+		self.buttons = []
+		for emoji_index in range(possible_emojis_size):
+			button = VoteButton(emojis[emoji_index])
+			button.set_embed_reference(ref_embed)
+			self.add_item(button)
+			self.buttons.append(button)
+
+	def disable_buttons(self) -> None:
+		for button in self.buttons:
+			button.disabled = True
+
+
+class PollEmbed(Embed):
```

Comment:
I don't think I can really do more separating than get_and_print_winner function. Because I made it so PollEmbed is 'self-sufficient' it updates(builds) itself and allow other classes/functions manipulate it to some extent. I am not sure if this is a good practice, but in this case it works quite well, and readability is also good. Yes, I do agree with get_and_print_winner function. PollEmbed is a like a poll object. I think if you would need to have multiple polls then it's gonna be actually better to have most business logic inside of the Embed, to give each one of them freedom. My opinion, correct me please if I am wrong.

Reply:
I will talk about the MVC architecture in the review. However, my point isn't that it needs to be "self-sufficient", it's that it handles stuff that it shouldn't. Embed isn't where you business logic should be. As I said, in other comments, you should have a model/class called Poll which would contain most of the information that this here (like the emojies, the poll counter, and even the title etc). The embed is simply a way to display information. That's all.